modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -193,11 +193,11 @@ class ReplCalculator(
         if (queue.nonEmpty) {
           //queue could be empty if InterruptRequest was asked!
           log.debug("Dequeuing execute request current size: " + queue.size)
-          queue = queue.dequeue._2
-          queue.headOption foreach { case (ref, er) =>
-            log.debug("About to execute request from the queue")
-            execute(ref, er)
-          }
+          val (executeRequest, newQueue) = queue.dequeue
+          queue = newQueue
+          val (ref, er) = executeRequest
```

Comment:
> So having a separate variable for the current executing request is probably better.

yeah, but this (so far) is not being used anyway, right? i.e. it should be good as is?